crates/app/src/providers.rs

```diff
@@ -15,7 +15,7 @@ use {
 #[derive(Clone)]
 pub struct StorageProvider {
     storage: Box<dyn Storage>,
-    namespace: Vec<u8>,
+    pub namespace: Vec<u8>,
```

Comment:
I think this should be kept private, and instead add a `.namespace()` method that returns a `&[u8]`.

This is because with this public, the namespace can be mutated, which isn't intended:

```rust
storage.namespace = vec![1, 2, 3, 4, 5];
```